cli/lsp/logging.rs

```diff
@@ -1,13 +1,90 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 
+use chrono::DateTime;
+use chrono::Utc;
+use deno_core::parking_lot::Mutex;
+use std::fs;
+use std::io::prelude::*;
+use std::path::PathBuf;
 use std::sync::atomic::AtomicBool;
 use std::sync::atomic::AtomicUsize;
 use std::sync::atomic::Ordering;
+use std::thread;
+use std::time::SystemTime;
 
 static LSP_DEBUG_FLAG: AtomicBool = AtomicBool::new(false);
 static LSP_LOG_LEVEL: AtomicUsize = AtomicUsize::new(log::Level::Info as usize);
 static LSP_WARN_LEVEL: AtomicUsize =
   AtomicUsize::new(log::Level::Warn as usize);
+static LOG_FILE: LogFile = LogFile {
+  enabled: AtomicBool::new(true),
+  path: Mutex::new(None),
+  buffer: Mutex::new(String::new()),
+};
+
+pub struct LogFile {
+  enabled: AtomicBool,
+  path: Mutex<Option<PathBuf>>,
+  buffer: Mutex<String>,
```

Comment:
```
  path: Mutex<Option<PathBuf>>,
  buffer: Mutex<String>,
```

The code is fine, but multiple mutexes increase the chance of deadlocks. To eliminate the question "will these two mutexes ever deadlock?" and reduce locking, can you modify the code to only use one Mutex?

Edit: Actually see my comment below. We can completely get rid of path being in a mutex.